addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java

```diff
@@ -273,12 +278,32 @@ private synchronized void refreshData() {
 	public void onConnectionLost() {
 		logger.info("Bridge connection lost. Updating thing status to OFFLINE.");
 		previousOnline = false;
-		updateStatus(ThingStatus.OFFLINE);
+		updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR);
+		for (Thing thing : getThing().getThings()) {
+			ThingHandler handler = thing.getHandler();
+			if (handler != null && handler instanceof MaxDevicesHandler ) {
+				((MaxDevicesHandler) handler).setOffLine();
```

Comment:
I actually think you are right, I think I may not need to set the
thingstatus to OFFLINE in the devicehandler as it will be already done via
the bridge OFFLINE action,
However, the main purpose of the .setOffLine() is to ensure I set the flag
to read/update all the values again when new data arrives (online again).
What I can do is to remove the thingstatus update to OFFLINE in this
.setOffLine()
if that helps

I found that in few combinations where I load/unload the binding,
delete/re-adding devices, online/offline of the cube it is not updating the
data properly.
With this PR I hope to tackle the update problems for most of these
somewhat unusual flows

2015-09-03 23:33 GMT+02:00 Kai Kreuzer <notifications@github.com>:

> In
> addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java
> <https://github.com/openhab/openhab2/pull/357#discussion_r38699350>:
>
> > @@ -273,12 +278,32 @@ private synchronized void refreshData() {
> >  	public void onConnectionLost() {
> >  		logger.info("Bridge connection lost. Updating thing status to OFFLINE.");
> >  		previousOnline = false;
> > -		updateStatus(ThingStatus.OFFLINE);
> > +		updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR);
> > +		for (Thing thing : getThing().getThings()) {
> > +			ThingHandler handler = thing.getHandler();
> > +			if (handler != null && handler instanceof MaxDevicesHandler ) {
> > +				((MaxDevicesHandler) handler).setOffLine();
>
> I would have thought that the status of child things is automatically set
> by the framework to OFFLINE, if a bridge goes OFFLINE. Is this not the case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab2/pull/357/files#r38699350>.
>